js/main.js

```diff
@@ -630,16 +656,270 @@ function showCluster(a) {
+    }
+
+    // Otherwise, we're doing our continuous variable search
+    else {
+        sigInst.iterNodes(function (a) {
+            var currentValue = Number(a.attr.attributes[attributeName]);
+            // Otherwise, we have a minimum and maximum value to consider for our search    
+            if(currentValue >= Number(minValue) && currentValue <= Number(maxValue))
+                outputList.push(a.id);
+        });
+    }
+
+    showClusterForAttribute(attributeName, outputList);
+}
+
+// Show cluster given the attribute "a" that you're looking for
+function showClusterForAttribute(a, b) {
+    if (b && 0 < b.length) {
+        showGroups(!1);
+        sigInst.detail = !0;
+        b.sort();
+        sigInst.iterEdges(function (a) {
+            a.hidden = !1;
+            a.attr.lineWidth = !1;
+            a.attr.color = !1
+        });
+        sigInst.iterNodes(function (a) {
+            a.hidden = !0
+        });
+        for (var f = [], e = [], c = 0, g = b.length; c < g; c++) {
+            //alert("b is " + b + " and c is " + c);
+            var d = sigInst._core.graph.nodesIndex[b[c]];
+            !0 == d.hidden && (e.push(b[c]), d.hidden = !1, d.attr.lineWidth = !1, d.attr.color = d.color, f.push('<li class="membership"><a href="#'+d.label+'" onmouseover="sigInst._core.plotter.drawHoverNode(sigInst._core.graph.nodesIndex[\'' + d.id + "'])\" onclick=\"nodeActive('" + d.id + '\')" onmouseout="sigInst.refresh()">' + d.label + "</a></li>"))
+        }
+        sigInst.clusters[a] = e;
+        sigInst.draw(2, 2, 2, 2);
+        // The next line needs to be changed so that we have a good description of the attribute we're searching for
+        // $GP.info_name.html("<b>" + sigInst.clusterNames[a] + "</b>");
+        $GP.info_name.html("<b>" + a + "</b>");
+        $GP.info_data.hide();
+        $GP.info_p.html("Group Members:");
+        $GP.info_link.find("ul").html(f.join(""));
+        $GP.info.animate({width:'show'},350);
+        //$GP.search.clean();
+        $GP.cluster.hide();
+        return !0
+    }
+    return !1
+}
+
+
+
+
+function populateSearch() { 
+    // THIS IS GENIE UPDATED
+    var firstElem;
+    for(elem in sigInst._core.graph.nodesIndex){
+        firstElem = elem;
+        break;
+    }
+    if(isNaN(firstElem))
+        firstElem = String(firstElem);
+    var b = sigInst._core.graph.nodesIndex[firstElem];
+    var f = b.attr;
+    
+    if (f.attributes) {
+        $("#variables").append(new Option("Phenotype Name", "Phenotype Name"));
+
+        for (var attr in f.attributes)
+            $("#variables").append(new Option(attr, attr));
+    }
+}
+
+
+function addExamplesForSearch(e){
+    var firstElem;
+    for(elem in sigInst._core.graph.nodesIndex){
+        firstElem = elem;
+        break;
+    }
+    if(isNaN(firstElem))
+        firstElem = String(firstElem);
+    var b = sigInst._core.graph.nodesIndex[firstElem];
+    
+    var f = b.attr;
+
```

Comment:
This code has repeated 3 times. Check if you can convert to another small function